### PR TITLE
[RFC] vim-patch:8.0.0131 and vim-patch:8.0.0155

### PR DESCRIPTION
**Pull request by ckelsel**
vim-patch:8.0.0131
    
    Problem:    Not enough test coverage for syntax commands.
    Solution:   Add more tests. (Dominique Pelle)
    
    https://github.com/vim/vim/commit/73b484c4da00011317dc68ada4f5dfc6515ad263
    
    vim-patch 8.0.0155
    
    Problem:    When sorting zero elements a NULL pointer is passed to qsort(),
                which ubsan warns for.
    Solution:   Don't call qsort() if there are no elements. (Dominique Pelle)
    
    https://github.com/vim/vim/commit/a216255a4faa91a15e7005ac319f2f62294f3f9e

### CODE REVIEW

**Comment by justinmk** [comment]
```
/home/travis/build/neovim/neovim/src/nvim/syntax.c:5849:9: runtime error: null pointer passed as argument 1, which is declared to never be null
/usr/include/stdlib.h:766:30: note: nonnull attribute specified here
    #0 0x17d599b in syntime_report /home/travis/build/neovim/neovim/src/nvim/syntax.c:5849:3
    #1 0x17d3400 in ex_syntime /home/travis/build/neovim/neovim/src/nvim/syntax.c:5762:5
    #2 0xb901d1 in do_one_cmd /home/travis/build/neovim/neovim/src/nvim/ex_docmd.c:2239:5
    #3 0xb6e913 in do_cmdline /home/travis/build/neovim/neovim/src/nvim/ex_docmd.c:607:20
    #4 0xb74be5 in do_cmdline_cmd /home/travis/build/neovim/neovim/src/nvim/ex_docmd.c:279:10
    #5 0x90c1a8 in f_execute /home/travis/build/neovim/neovim/src/nvim/eval.c:8015:5
    #6 0x817cc3 in call_func /home/travis/build/neovim/neovim/src/nvim/eval.c:6408:11
    #7 0x82d5c3 in get_func_tv /home/travis/build/neovim/neovim/src/nvim/eval.c:6148:11
    #8 0x8b9b16 in eval7 /home/travis/build/neovim/neovim/src/nvim/eval.c:4248:15
    #9 0x8b603b in eval6 /home/travis/build/neovim/neovim/src/nvim/eval.c:3945:7
    #10 0x8b3947 in eval5 /home/travis/build/neovim/neovim/src/nvim/eval.c:3796:7
    #11 0x8aec4f in eval4 /home/travis/build/neovim/neovim/src/nvim/eval.c:3533:7
    #12 0x8ae12c in eval3 /home/travis/build/neovim/neovim/src/nvim/eval.c:3451:7
    #13 0x8ad5fc in eval2 /home/travis/build/neovim/neovim/src/nvim/eval.c:3382:7
    #14 0x8101e2 in eval1 /home/travis/build/neovim/neovim/src/nvim/eval.c:3309:7
    #15 0x80f5fa in eval0 /home/travis/build/neovim/neovim/src/nvim/eval.c:3269:9
    #16 0x81c0db in ex_let /home/travis/build/neovim/neovim/src/nvim/eval.c:1499:9
    #17 0xb901d1 in do_one_cmd /home/travis/build/neovim/neovim/src/nvim/ex_docmd.c:2239:5
    #18 0xb6e913 in do_cmdline /home/travis/build/neovim/neovim/src/nvim/ex_docmd.c:607:20
    #19 0x84d87d in call_user_func /home/travis/build/neovim/neovim/src/nvim/eval.c:21337:3
    #20 0x8175e7 in call_func /home/travis/build/neovim/neovim/src/nvim/eval.c:6393:11
    #21 0x82d5c3 in get_func_tv /home/travis/build/neovim/neovim/src/nvim/eval.c:6148:11
    #22 0x825d3b in ex_call /home/travis/build/neovim/neovim/src/nvim/eval.c:2760:9
    #23 0xb901d1 in do_one_cmd /home/travis/build/neovim/neovim/src/nvim/ex_docmd.c:2239:5
    #24 0xb6e913 in do_cmdline /home/travis/build/neovim/neovim/src/nvim/ex_docmd.c:607:20
    #25 0x870405 in ex_execute /home/travis/build/neovim/neovim/src/nvim/eval.c:19520:7
    #26 0xb901d1 in do_one_cmd /home/travis/build/neovim/neovim/src/nvim/ex_docmd.c:2239:5
    #27 0xb6e913 in do_cmdline /home/travis/build/neovim/neovim/src/nvim/ex_docmd.c:607:20
    #28 0x84d87d in call_user_func /home/travis/build/neovim/neovim/src/nvim/eval.c:21337:3
    #29 0x8175e7 in call_func /home/travis/build/neovim/neovim/src/nvim/eval.c:6393:11
    #30 0x82d5c3 in get_func_tv /home/travis/build/neovim/neovim/src/nvim/eval.c:6148:11
    #31 0x825d3b in ex_call /home/travis/build/neovim/neovim/src/nvim/eval.c:2760:9
    #32 0xb901d1 in do_one_cmd /home/travis/build/neovim/neovim/src/nvim/ex_docmd.c:2239:5
    #33 0xb6e913 in do_cmdline /home/travis/build/neovim/neovim/src/nvim/ex_docmd.c:607:20
    #34 0xb5b140 in do_source /home/travis/build/neovim/neovim/src/nvim/ex_cmds2.c:2935:3
    #35 0xb57604 in cmd_source /home/travis/build/neovim/neovim/src/nvim/ex_cmds2.c:2680:14
    #36 0xb57730 in ex_source /home/travis/build/neovim/neovim/src/nvim/ex_cmds2.c:2661:3
    #37 0xb901d1 in do_one_cmd /home/travis/build/neovim/neovim/src/nvim/ex_docmd.c:2239:5
    #38 0xb6e913 in do_cmdline /home/travis/build/neovim/neovim/src/nvim/ex_docmd.c:607:20
    #39 0xb74be5 in do_cmdline_cmd /home/travis/build/neovim/neovim/src/nvim/ex_docmd.c:279:10
    #40 0xeb8ecb in exe_commands /home/travis/build/neovim/neovim/src/nvim/main.c:1644:5
    #41 0xea6ba2 in main /home/travis/build/neovim/neovim/src/nvim/main.c:506:5
    #42 0x2ae6b0f1ff44 in __libc_start_main /build/eglibc-SvCtMH/eglibc-2.19/csu/libc-start.c:287
    #43 0x44d22b in _start (/home/travis/build/neovim/neovim/build/bin/nvim+0x44d22b)

SUMMARY: AddressSanitizer: undefined-behavior /home/travis/build/neovim/neovim/src/nvim/syntax.c:5849:9 in 
```

**Comment by ckelsel** [comment]
@justinmk I don't know how to fix this. It's a little difficult for me.

**Comment by justinmk** [comment]
No problem. I or someone else will look at it.

**Comment by oni-link** [comment]
Test failure because of missing vim-patch:8.0.0155.

File: src/nvim/syntax.c [hunk 1]
```diff
@@ -5845,9 +5845,12 @@ static void syntime_report(void)
     }
   }
 
-  /* sort on total time */
-  qsort(ga.ga_data, (size_t)ga.ga_len, sizeof(time_entry_T),
-      syn_compare_syntime);
+  // Sort on total time. Skip if there are no items to avoid passing NULL
+  // pointer to qsort().
```
Comment (oni-link) [inline src/nvim/syntax.c]: Skip ... -> Skip if there is nothing to sort.

**Pull request closed without merging**